packages/gatsby-react-router-scroll/src/ScrollBehaviorContext.ts

```diff
@@ -14,25 +17,44 @@ const childContextTypes = {
   scrollBehavior: PropTypes.object.isRequired,
 }
 
-class ScrollContext extends React.Component {
-  constructor(props, context) {
+interface IProps {
+  shouldUpdateScroll: ShouldUpdateScroll<IContext>
+  children: React.ReactElement
+  location: LocationBase
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface IContext {}
+
+class ScrollContext extends React.Component<IProps> {
+  static propTypes: PropTypes.InferType<typeof propTypes>
+  static childContextTypes: PropTypes.InferType<typeof childContextTypes>
+
+  scrollBehavior: ScrollBehavior<LocationBase, IContext>
+
+  constructor(props: IProps, context: IContext) {
     super(props, context)
 
     this.scrollBehavior = new ScrollBehavior({
       addTransitionHook: history.listen,
+      // Type definition for LocationBase from scroll-behavior should extend
+      // from Window['location'] for it to be compatible with WindowLocation
+      // from @reach/router
+      // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+      // @ts-ignore
       stateStorage: new SessionStorage(),
-      getCurrentLocation: () => this.props.location,
+      getCurrentLocation: (): LocationBase => this.props.location,
       shouldUpdateScroll: this.shouldUpdateScroll,
     })
   }
 
-  getChildContext() {
+  getChildContext(): PropTypes.InferType<typeof childContextTypes> {
```

Comment:
I would prefer not to have this complex types when this could be relatively simple `{ scrollBehavior: typeof ScrollContext }`